### ISSUE TITLE
New legend disposer causes crash

### ISSUE THREAD

**Issue by saveriocaminitiktech** [amcharts/amcharts4#2632]
**Bug description**

In version 4.9.25 @martynasma introduced
```
legend.addDisposer(new Disposer(() => {
    this.legend = undefined;
}))
```
in /src/.internal/charts/Chart.ts:354.

This is clearly in contrast with /src/.internal/charts/types/MapChart.ts:2218
```
protected setLegend(legend: Legend) {
  super.setLegend(legend);
  legend.parent = this;
}
```
So whenever i call .dispose() on my maps I get a TypeError: Cannot set property 'parent' of undefined.

**Additional context**
I have a React app and call map.dispose on componentWillUnmount. This bug makes my entire app crash.

**Proposed solution**
Simply change MapChart.ts:2220 and make it more robust with a check on legend. Also if undefined is a legit value for setLegend I suggest to properly annotare the function pararm.
```
protected setLegend(legend?: Legend) {
  super.setLegend(legend);
  legend && legend.parent = this;
}
```

**Comment by zeroin** [comment]
Thanks for noticing, we will fix this in next release.

**Comment by martynasma** [comment]
Fixed in [4.9.28](https://github.com/amcharts/amcharts4/releases/tag/4.9.28).

## [4.9.28] - 2020-06-29

### Added
- New adapter `xlsxWorkbook` in `Export`. Will allow decorating a Workbook object before it is exported.
- `chart.exporting` text-based API functions (`getSVG()`, `getCSV()`, `getJSON()`, `getHTML()`) now have third parameter (boolean). If set to `false` they will return a raw output, rather than data:uri.
- `zoomable` property added to all axes (default: `true`). Setting it to `false` will make axis not react to zoom events from scrollbar/cursor.

### Changed
- Removed `focusable` from map chart background series.
- Polyfills in `.js` file will now only load on-demand, making it with some frameworks that override default objects.

### Fixed
- Regression plugin: the regression line was not being drawn correctly on a chart with horizontal `ValueAxis`.
- Export: images exported on zoomed-out pages sometimes were showing visual artifacts.
- If a chart cursor was visible (because of soft or hard `triggerMove()` calls) but the real mouse was out of the plot area, the cursor used to react to clicks even though it shouldn't have.
- `ForceDirectedNode`'s `fill` and `stroke` adapters were not working properly if nodes were initially collapsed.
- `Sprite.hitTest()` method was not working properly with `nonScaling = true`.
- If a `MapChart` had geodata loaded via `geodataSource.url`, the map used to disappear after div size changed.
- If a legend of a `MapChart` was disposed, a JS error might happen in some cases.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
